client/src/components/PacmanGame/__tests__/gameCore.test.js

```diff
@@ -0,0 +1,78 @@
+import {
+  initSquareGridState,
+  entityToCode,
+  codeToEntity,
+  entityApplier,
+} from '../gameCore';
+
+
+describe('initSquareGridState', () => {
+  const threeZeroGrid = [0, 0, 0];
+  const threeByThreeGrid = [[...threeZeroGrid], [...threeZeroGrid], [...threeZeroGrid]];
+  it('should generates the square grid', () => {
+    expect(initSquareGridState(3)).toEqual(threeByThreeGrid);
+  });
+  it('should generates 2d array with correct number of rows', () => {
+    expect(initSquareGridState(3).length).toEqual(3);
+  });
+  it('should generates 2d array with correct number of cols', () => {
+    expect(initSquareGridState(3).length).toEqual(3);
+  });
+});
+
+describe('codeToEntity', () => {
+  it('should be free for code 0', () => {
+    expect(codeToEntity('0')).toEqual('free');
+  });
+  it('should be food for code 1', () => {
+    expect(codeToEntity('1')).toEqual('food');
+  });
+  it('should be free for code 2', () => {
+    expect(codeToEntity('2')).toEqual('energizer');
+  });
+  it('should be free for code 3', () => {
+    expect(codeToEntity('3')).toEqual('ghost');
+  });
+  it('should be free for code 4', () => {
+    expect(codeToEntity('4')).toEqual('wall');
+  });
+  it('should be free for code 5', () => {
+    expect(codeToEntity('5')).toEqual('pacman');
+  });
```

Comment:
This is not required. It is just extracting the key from the object. It is basically testing the object functionality of JS.
And the message passed in 'it' block is not correct. It is 'free' for all except one.

Reply:
@318097 
I'll correct the text under `it`.

The reason for these testing 2 functions(`codeToEntity` and `entityToCode`) is that these will be essential for all the game logic functionalities. Later, if someone decides to tweak/optimize/change the inner implementation of these 2 functions, the outer interface should not change. 
**If the outer interface changes, these 2 tests will fail there and then.**